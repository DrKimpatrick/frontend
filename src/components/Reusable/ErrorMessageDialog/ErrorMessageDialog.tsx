import React, { FC } from 'react';
import InfoIcon from '@material-ui/icons/Info';

interface MessageDialogProps {
  title: string;
  message: string;
}

const ErrorMessageDialog: FC<MessageDialogProps> = ({ title, message }) => {
  return (
    <div className="w-full mt-4 flex flex-col items-center">
      <div className="mb-4">
        <span className="text-red-600">
          <InfoIcon color="inherit" style={{ fontSize: 60 }} />
        </span>
      </div>
      <div className="mb-4 flex">
        <span className="text-2xl text-center text-red-600">{title}</span>
      </div>
      <div className="mb-4 flex">
        <span className="text-sm text-center">{message}</span>
      </div>
    </div>
  );
};

export default ErrorMessageDialog;
