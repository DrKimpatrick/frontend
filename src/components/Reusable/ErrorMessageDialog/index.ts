import ErrorMessageDialog from './ErrorMessageDialog';

export { ErrorMessageDialog };
