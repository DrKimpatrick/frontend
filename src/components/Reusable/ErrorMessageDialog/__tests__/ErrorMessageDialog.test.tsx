import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ErrorMessageDialog from '../ErrorMessageDialog';

afterEach(cleanup);

const props = {
  title: 'We have a message',
  message: 'something went terribly wrong'
};

it('renders snapshot successfully', () => {
  const container = render(<ErrorMessageDialog {...props} />);
  expect(container).toMatchSnapshot();
});
