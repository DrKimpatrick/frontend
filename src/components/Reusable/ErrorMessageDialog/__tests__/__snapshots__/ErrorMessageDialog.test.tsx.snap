// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full mt-4 flex flex-col items-center"
      >
        <div
          class="mb-4"
        >
          <span
            class="text-red-600"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              style="font-size: 60px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="mb-4 flex"
        >
          <span
            class="text-2xl text-center text-red-600"
          >
            We have a message
          </span>
        </div>
        <div
          class="mb-4 flex"
        >
          <span
            class="text-sm text-center"
          >
            something went terribly wrong
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full mt-4 flex flex-col items-center"
    >
      <div
        class="mb-4"
      >
        <span
          class="text-red-600"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="font-size: 60px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
        </span>
      </div>
      <div
        class="mb-4 flex"
      >
        <span
          class="text-2xl text-center text-red-600"
        >
          We have a message
        </span>
      </div>
      <div
        class="mb-4 flex"
      >
        <span
          class="text-sm text-center"
        >
          something went terribly wrong
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
